frontend/insurance/src/app/core/services/middleware/middleware.service.spec.ts

```diff
@@ -2,19 +2,56 @@ import { TestBed } from '@angular/core/testing';
 
 import { MiddlewareService } from './middleware.service';
 import { AuthService } from '../auth/auth.service';
-import { RouterTestingModule } from '@angular/router/testing';
 import { HttpClientModule } from '@angular/common/http';
+import { RouterTestingModule } from '@angular/router/testing';
+import { of } from 'rxjs';
+
+describe('MiddlewareService', () => {
+  let service: MiddlewareService;
 
-describe('MidwareService', () => {
-  beforeEach(() =>
+  beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [RouterTestingModule, HttpClientModule],
+      imports: [HttpClientModule, RouterTestingModule],
       providers: [AuthService],
-    }),
-  );
+    });
+
+    service = TestBed.get(MiddlewareService);
+  });
 
   it('should be created', () => {
-    const service: MiddlewareService = TestBed.get(MiddlewareService);
     expect(service).toBeTruthy();
   });
+
+  describe('getClaims', () => {
+    it('should return an array of claims', () => {
+      const claimsResponse: Array<Protos.Claim> = [
+        {
+          status: 0,
+          created: '',
+          modified: '',
+          date_of_loss: '',
+          vehicle: {
+            vin: '1J4GA59178L553429',
+            miles: 123,
+            location: '',
+          },
+          insurer: {
+            name: '',
+            deductible: 1,
+            has_gap: false,
+          },
+          files: [],
+        },
+      ];
+
+      let response;
+      spyOn(service, 'getClaims').and.returnValue(of(claimsResponse));
+
+      service.getClaims().subscribe(res => {
+        response = res;
+      });
+
+      expect(response).toEqual(claimsResponse);
```

Comment:
This test is useless. You should be testing you **_UI_** to make sure it correctly takes the response and uses it appropriately.

You're effectively just testing karma here.

Reply:
This is testing to make sure the data from the get matches the expected format, no?

Reply:
(but yes, i will be adding the component tests that gets the claims)